### PR TITLE
Added instructions to run `sos report` for EDPM compute nodes

### PR DESCRIPTION
**Pull request by rebtoor**
Also: 
- Fixed markdown errors

Depends-On: https://github.com/sosreport/sos/pull/3414
JIRA [OSPRH-5744](https://issues.redhat.com/browse/OSPRH-5744)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -169,14 +172,31 @@ The command above will create three pods associated to the existing `imagestream
...
 [must-gather] OUT pod for plug-in image quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e9601b492cbb375f0a05310efa6025691f8bba6a97667976cd4baf4adf0f244c created
 ...
 ...
 ```
 
+### EDPM compute must gather via `sos` tool
+
+To gather config files and logs from the deployed compute nodes, we need to rely on the `sos` tool (former `sosreport`).
+
+There's an open [PR](https://github.com/sosreport/sos/pull/3414) to enable path gathering from EDPM deployed compute nodes so if you want to test it on running nodes, you need to follow these steps on a running compute node:
```
Comment (fmount) [inline README.md:187]: I would rather add a "Depends-On" (a logical depends on given we don't have any CI here), and wait until the patch [1] merges.
When [1] merges, we can change this statement to describe how to run a standalone `sos report` on an existing compute node (which is basically described below).

[1]  https://github.com/sosreport/sos/pull/3414

File: README.md [hunk 2]
```diff
@@ -169,14 +172,31 @@ The command above will create three pods associated to the existing `imagestream
...
+
+```bash
+cd ~
+git clone https://github.com/rebtoor/sos.git -b osp18
```
Comment (fmount) [inline README.md:191]: As long as [1] merges, you can git clone `https://github.com/sosreport/sos` on the main branch

[1] https://github.com/sosreport/sos/pull/3414

File: README.md (hunk 1 above)
Comment (rebtoor) [reply README.md:187]: Totally agree on this.

File: README.md (hunk 2 above)
Comment (rebtoor) [reply README.md:191]: I guess that we should remove the clone step, since we expect that sos will be already installed on compute nodes. WDYT?

Comment (fmount) [reply README.md:191]: +1, and we can provide instructions to run the sosreport both from a coreos node and from an edpm one using the new profile

File: README.md [hunk 3]
```diff
@@ -68,21 +71,21 @@ In the above case, where the namespace is `openshift-must-gather-b9fpw` and the
...
 ```bash
-$ oc -n openshift-must-gather-b9fpw rsh must-gather-mxc7k
+oc -n openshift-must-gather-b9fpw rsh must-gather-mxc7k
 ```
 
 And then if we were debugging a bash script called `gather_trigger_gmr`, we
 would run it in debug mode:
 
 ```bash
-sh-5.1# bash -x /usr/bin/gather_trigger_gmr
```
Comment (Akrog) [inline README.md:78]: -1: This prefix was written on purpose to reflect that this is running within the container.

File: README.md [hunk 4]
```diff
@@ -169,14 +172,142 @@ The command above will create three pods associated to the existing `imagestream
...
 [must-gather] OUT pod for plug-in image quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e9601b492cbb375f0a05310efa6025691f8bba6a97667976cd4baf4adf0f244c created
 ...
 ...
 ```
 
+### `sos report` from nodes
+
+To gather config files and logs from hosts we need to rely on the `sos` tool (former `sosreport`).
+
+We can obtain logs from both OCP cluster nodes and compute nodes.
```
Comment (Akrog) [inline README.md:187]: For OCP cluster nodes the SOS report should be gathered using the mechanism that is being added in [PR#19](https://github.com/openstack-k8s-operators/openstack-must-gather/pull/19), though I think it has value to document the manual mechanism here.

File: README.md [hunk 5]
```diff
@@ -169,14 +172,142 @@ The command above will create three pods associated to the existing `imagestream
...
+Register this system with Red Hat Insights: insights-client --register
+Create an account or view all your systems at https://red.ht/insights-dashboard
+Last login: Wed Nov 15 10:02:11 2023 from 192.168.111.1
+[zuul@compute-0 ~]$
+```
+
+And launch the `sos` command just like we did in the other example, but with different flags.
+
+```bash
+sudo sos report -p system,virt,openstack_edpm
```
Comment (Akrog) [inline README.md:304]: -1: This is insufficient, we also need storage related plugins such as `block,iscsi,scsi,multipath,devicemapper,devices,lvm2,nfs,cifs,nvme,udev`

File: README.md [hunk 6]
```diff
@@ -169,14 +172,142 @@ The command above will create three pods associated to the existing `imagestream
...
+```
+
+As you can see in the example above, there's a new profile for the `sos` tool, `openstack_edpm`, which enables the gathering from the plugins running on the node.
```
Comment (Akrog) [inline README.md:307]: For other reviewers, [that is being added in this PR](https://github.com/openstack-k8s-operators/openstack-must-gather/pull/18)

**Review by Akrog — CHANGES_REQUESTED** [review]
Thank you for the documentation effort. :-)

File: README.md (hunk 5 above)
Comment (rebtoor) [reply README.md:304]: ack!

File: README.md (hunk 6 above)
Comment (Akrog) [reply README.md:307]: @fmount Should we wait until the `sos` PR gets merged before merging this one?

Comment (fmount) [reply README.md:307]: Yeah, this is the idea, we can publish the `README` after that patch merges (I'm not sure how the building/shipping process works on the `sos` project, so we might have to wait a bit more).
In general I agree to hold this one until https://github.com/sosreport/sos/pull/3414 lands.

Comment (Akrog) [reply README.md:307]: Following these instructions I have this patch in my local repo to do this upon request with must-gather: https://github.com/Akrog/openstack-must-gather/commit/c0fc8872727dcba84b68bd8ba7b0e3f65e843dd0

**Comment by rebtoor** [comment]
@fmount as you can see sos patch has been merged, but there's no new release of sos (yet). I'd wait to merge this PR until we'll have a new release of sos.

**Review by fmount — APPROVED** [review]
/lgtm

**Pull request merged**